### ISSUE TITLE
DatePicker: missing `role="dialog"` and `aria-modal="true"`

### ISSUE THREAD

**Issue by nikitaorliak-cengage** [cengage/react-magma#2471]
We probably need `role="dialog"` and `aria-modal="true"` in our DatePicker calendar for accessibility improvement. Needs investigation.
Also, we need to verify the behavior of the nested `aria-modal` element within the `Modal`, and also determine whether the built-in rendering (without the portal) causes screen readers to continue reading adjacent content.

**Comment by artemtrusov-cengage** [comment]
This ticket also fixed this issue: https://github.com/cengage/react-magma/issues/2568

**Comment by serhiiyanhol-cengage** [comment]
Tested on:

- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--inside-modal
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/datepicker--inside-modal

Everything works as expected. No bugs found.
